### PR TITLE
feat: added game Play

### PR DESCRIPTION
**Pull request by amitojsingh**
1. Edit Database folder and stored all the database operations in it. 
2. Made Start button disabled until the optionselected 
3. Added Types for shabadavali database
4. Refactored Code
5. Implemented Game Play
6. implemented  Words Learnt
7. Store Words Learning
8. update the progress

### CODE REVIEW

File: src/pages/dashboard/utils/index.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import {
+  GameScreen,
+  QuestionType,
+  User,
+  WordShabadavaliDB,
+} from 'types/shabadavlidb';
+// import { getRandomQuestion } from 'database/default/question';
+// import ALL_CONSTANT from 'constants/constant';
```
Comment (MySmilingTurban) [inline src/pages/dashboard/utils/index.ts:8]: remove comments if not needed

File: src/routes/index.tsx [hunk 2]
```diff
@@ -10,7 +10,7 @@ import Examples from 'pages/word/examples';
 import WordsPageLayout from 'pages/word/layout';
 import NotFound from 'pages/not-found';
 import Home from 'pages/page';
-import Semantics from 'pages/word/semantics';
+// import Semantics from 'pages/word/semantics';
```
Comment (MySmilingTurban) [inline src/routes/index.tsx:13]: are we not showing semantics (synonyms/antonyms) page anymore??

**Review by MySmilingTurban — CHANGES_REQUESTED** [review]
Veerji, the image is taking significantly higher timing to load, everything else seems to work fine.
Suggested some minor changes

File: src/database/shabadavalidb/questions.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+) => {
+  // const randomID = generateRandomId();
+  const questionCollectionRef = getQuestionCollectionRef(uid);
+  const queryRef = query(
+    questionCollectionRef,
+    where('isLearnt', '==', isLearnt),
+    // where('id', '<=', randomID),
+    limit(count),
+  );
+  const questionSnapshots = await getDocs(queryRef);
```
[the remark this answers is not in the document]
Comment (amitojsingh) [reply src/database/shabadavalidb/questions.ts:86]: not actually. but I am shuffling after the result. 
The approach which we are using like generating random ID. It has some limitation. The main limitation is 
This only queries a portion of the data (everything less than or equal to the random ID) we might miss out on a significant portion of your collection.
However, I got null values many times so I came up this for now. we can improve.

File: src/database/shabadavalidb/users.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+export const updateUser = async (
+  userRef: DocumentReference,
+  userData: UserProfile,
+) => {
+  const updatedUser = await setDoc(userRef, {
+    ...userData,
+  });
+  return updatedUser;
+};
```
[the remark this answers is not in the document]
Comment (amitojsingh) [reply src/database/shabadavalidb/users.tsx:41]: I have removed looks like we are not using it anywhere.

File: src/routes/index.tsx (hunk 2 above)
Comment (amitojsingh) [reply src/routes/index.tsx:13]: Not now, I can not see any questionType named Semantics. So it would be totally different approach.

Comment (MySmilingTurban) [reply src/routes/index.tsx:13]: It is a not a question type, it is a part of word description and is not marked (drag-drop is used to make it interactive)

**Comment by amitojsingh** [comment]
Okay let’s make a new task/issue for this. I’ll add it in coming week.

*There is no substitue of hardwork.*
Thomas A Edison

Amitoj singh
Full Stack Developer
M: +1 (647) 675-5277
https://github.com/amitojsingh

File: src/routes/index.tsx (hunk 2 above)
Comment (amitojsingh) [reply src/routes/index.tsx:13]: Okay let’s make a new separate task/issue for this. I’ll add it in the coming week.

**Review by MySmilingTurban — APPROVED** [review]
looks good to me!

**Pull request merged**